### PR TITLE
Asset repository

### PR DESCRIPTION
**Pull request by FairyTalePol**
Added repositories:
Asset, Private asset, Public asset, Asset-User relations, Asset-Group relations
Modified database and added tests

### CODE REVIEW

File: src/test/kotlin/com/example/springrestuser/repositories/AssetTest.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import org.testcontainers.containers.PostgreSQLContainer
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+import org.testcontainers.utility.DockerImageName
+
+@Testcontainers
+@SpringBootTest
+@AutoConfigureTestEntityManager
+@Transactional
+
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/repositories/AssetTest.kt:21]: empty line

File: src/test/kotlin/com/example/springrestuser/repositories/AssetTest.kt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import org.springframework.test.context.DynamicPropertyRegistry
+import org.springframework.test.context.DynamicPropertySource
+import org.springframework.transaction.annotation.Transactional
+import org.testcontainers.containers.PostgreSQLContainer
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+import org.testcontainers.utility.DockerImageName
+
+@Testcontainers
+@SpringBootTest
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/repositories/AssetTest.kt:18]: Why its @SpringBootTest?

File: src/test/kotlin/com/example/springrestuser/repositories/AssetTest.kt [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    fun `getting asset`() {
+        //Given
+        var expectedAsset = AssetEntity(15.0, "General Motors")
+        em.persist(expectedAsset)
+
+        //When
+        val asset = repo.findById(expectedAsset.id!!)
+
+        //Then
+        Assertions.assertEquals(expectedAsset, asset.orElse(AssetEntity(15.0, "General Motors")))
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/repositories/AssetTest.kt:94]: Please dont use static class import. Use static method import instead.
```
//Then
assertEquals(expectedAsset, asset.orElse(AssetEntity(15.0, "General Motors")))
```

File: src/test/kotlin/com/example/springrestuser/repositories/AssetTest.kt [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    @Test
+    fun `modifying asset`() {
+        //Given
+        var expectedAsset = AssetEntity(15.0, "General Motors")
+
+        //When
+        em.persist(expectedAsset)
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/repositories/AssetTest.kt:103]: Better to em.flush() to make sure you actually insert and them update

File: src/test/kotlin/com/example/springrestuser/repositories/GroupAssetRelationRepositoryTest.kt [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+import org.testcontainers.containers.PostgreSQLContainer
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+import org.testcontainers.utility.DockerImageName
+import javax.persistence.EntityManager
+import javax.persistence.PersistenceContext
+import javax.swing.GroupLayout
+
+@Testcontainers
+@SpringBootTest(
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/repositories/GroupAssetRelationRepositoryTest.kt:23]: Such test should be @DataJpaTest

File: src/test/kotlin/com/example/springrestuser/repositories/PrivateAssetTest.kt [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+import org.testcontainers.utility.DockerImageName
+
+@Testcontainers
+
+@SpringBootTest
+@AutoConfigureTestEntityManager
+@Transactional
+
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/repositories/PrivateAssetTest.kt:27]: empty lines

File: src/test/kotlin/com/example/springrestuser/repositories/PublicAssetTest.kt [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+    @BeforeEach
+    fun beforeEach()
+    {
+        repo.deleteAll()
+    }
+
+    @AfterEach
+    fun afterEach()
+    {
+        try{
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/repositories/PublicAssetTest.kt:74]: space after try please

File: src/test/kotlin/com/example/springrestuser/repositories/PublicAssetTest.kt [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+    @Test
+    fun `getting public asset by asset id`() {
+        //Given
+        val asset = AssetEntity(15.0, "General Motors")
+        em.persist(asset)
+        val publicAsset = PublicAssetEntity(asset,AssetType.A.toString(), "party na prykarpatti")
+        em.persist(publicAsset)
+        val expectedAsset= listOf(publicAsset)
+        //When
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/repositories/PublicAssetTest.kt:135]: Please add empty line before each block Given/When/Then

File: src/test/kotlin/com/example/springrestuser/repositories/UserAssetRelationRepositoryTest.kt [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+            registry.add("spring.datasource.password", container::getPassword)
+            registry.add("spring.datasource.username", container::getUsername)
+        }
+
+    }
+
+    @Test
+    fun containerStartTest() {
+        Assertions.assertTrue(container.isRunning)
+
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/repositories/UserAssetRelationRepositoryTest.kt:72]: empty line

File: src/test/kotlin/com/example/springrestuser/repositories/UserGroupTests.kt [hunk 10]
```diff
@@ -180,7 +180,7 @@ class UserGroupTests {
         em.persist(userGroup5)
 
 
-        val actual = repo.findByUserIdAndGroupId(user2.id?:0, group2.id)
+        val actual = repo.findByUserIdAndGroupId(user2.id?:0, group2.id?:0)
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/repositories/UserGroupTests.kt:183]: please add spaces

**Review by sn3jkeee — CHANGES_REQUESTED** [review]

File: src/test/kotlin/com/example/springrestuser/repositories/AssetTest.kt (hunk 2 above)
Comment (FairyTalePol) [reply src/test/kotlin/com/example/springrestuser/repositories/AssetTest.kt:18]: Because Flyway requires Sprig Boot cotext to perform migrations

**Review by sn3jkeee — APPROVED** [review]

**Pull request merged**